tools/devops/device-tests/scripts/VSTS.Tests.ps1

```diff
@@ -90,4 +90,92 @@ Describe 'Stop-Pipeline' {
             Assert-MockCalled -CommandName Invoke-RestMethod -Times 0 -Scope It 
         }
     }
-}
\ No newline at end of file
+}
+
+Describe 'Set-PipelineResult' {
+    Context 'with all the env vars present' {
+
+        BeforeAll {
+            $Script:envVariables = @{
+                "SYSTEM_TEAMFOUNDATIONCOLLECTIONURI" = "SYSTEM_TEAMFOUNDATIONCOLLECTIONURI";
+                "SYSTEM_TEAMPROJECT" = "SYSTEM_TEAMPROJECT";
+                "BUILD_BUILDID" = "BUILD_BUILDID";
+                "SYSTEM_ACCESSTOKEN" = "SYSTEM_ACCESSTOKEN"
+            }
+
+            $envVariables.GetEnumerator() | ForEach-Object { 
+                $key = $_.Key
+                Set-Item -Path "Env:$key" -Value $_.Value
+            }
+        }
+
+        It 'performs the rest call' {
```

Comment:
pwsh testing is bizarre! 🙈

Reply:
I think it was inspired by mocca and other js testing